src/main/java/uk/gov/hmcts/reform/jobscheduler/services/jobs/TriggerConverter.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.hmcts.reform.jobscheduler.services.jobs;
+
+import uk.gov.hmcts.reform.jobscheduler.model.Trigger;
+
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Date;
+
+import static org.quartz.TriggerBuilder.newTrigger;
+
+final class TriggerConverter {
+
+    private TriggerConverter() {
+        // static class constructor
+    }
+
+    static org.quartz.Trigger toQuartzTrigger(final Trigger trigger) {
+        return newTrigger()
+            .startAt(Date.from(trigger.startDateTime.toInstant()))
+            .usingJobData(JobDataKeys.ATTEMPT, 1)
```

Comment:
I'm not sure the converter should be concerned about attempt. What was the reason for moving it from the job?

Reply:
The flag `attempt` is relevant to trigger. In the FailListener the new trigger is scheduled with incremented attempt which is actually stored in trigger's data map, not the job's data map. It is essentially clearing up sequential trigger attempt as per our current implementation and not misusing data map values. Shortly:

- separate/distinguish data map for correct usage
- make logical trigger filtering

Reply:
Fair enough